Sources/RemindersLibrary/Reminders.swift

```diff
@@ -79,6 +133,52 @@ public final class Reminders {
         }
     }
 
+    func exportAllReminders(prettyPrint: Bool) {
+        let semaphore = DispatchSemaphore(value: 0)
+        let encoder = JSONEncoder()
+
+        if prettyPrint {
+            encoder.outputFormatting = .prettyPrinted
+        }
+
+        self.reminders(on: self.getCalendars(), displayOptions: .incomplete) { reminders in
+            let remindersByCalendarTitle = Dictionary(grouping: reminders) { reminder -> String in
+                return reminder.calendar?.title ?? "Unknown"
+            }
+
+            let mappedRemindersByCalendarTitle = remindersByCalendarTitle.mapValues { values in
+                values.map { reminder -> ReminderData in
+                    return ReminderData(
+                        id: reminder.calendarItemIdentifier,
+                        calendarTitle: reminder.calendar?.title,
+                        title: reminder.title,
+                        creationDate: reminder.creationDate,
+                        lastModifiedDate: reminder.lastModifiedDate,
+                        startDate: reminder.startDateComponents?.date,
+                        dueDate: reminder.dueDateComponents?.date,
+                        notes: reminder.notes,
+                        priority: reminder.priority,
+                        isCompleted: reminder.isCompleted,
+                        completionDate: reminder.completionDate,
+                        alarms: reminder.alarms,
+                        recurrenceRules: reminder.recurrenceRules
+                    )
+                }
+            }
+
+            do {
+                let data = try encoder.encode(mappedRemindersByCalendarTitle)
+                FileHandle.standardOutput.write(data)
+            } catch {
+                print(error)
+            }
```

Comment:
We might potentially want to let this error actually crash the program, or otherwise be handled in a more 'actually errored' sort of way?

Reply:
definitely needs to exit non-zero in this case

Reply:
Yup, makes sense. Will see if/what the established pattern is in the rest of the code and go from there.